EIPS/eip-draft_transaction_data_opcodes.md

```diff
@@ -0,0 +1,106 @@
+
+### Compiler Specification
+
+High-level programming languages that compile to EVM bytecode should provide support for these new instructions from a high-level perspective by introducing new members to the transactional context they already support for the `ORIGIN` instruction.
+
+## Rationale
+
+### Naming Conventions
+
+The `ORIGIN`-prefixed instructions attempted to conform to the existing naming convention of `CALL`-prefixed instructions given the existence of the `ORIGIN` instruction which is equivalent to the `CALLER` (`0x33`) instruction but on the original transaction's context.
+
+The `ENTRYPOINT` instruction came to be by defining a sensible name that immediately and clearly depicts what it is meant to achieve by signaling the first interaction with the blockchain, i.e. the entry-point.
+
+### Instruction Address Space
+
+The instruction address space of the `0x30-0x3f` has been exhausted by calls that already provide information about the execution context of a call so a new range had to be identified that is suitable for the purposes of the EIP.
+
+Given that the [EIP-1344](https://eips.ethereum.org/EIPS/eip-1344) `CHAINID` opcode was included at `0x46`, it made sense to include additional transaction-related data beyond it since the Chain ID is also included in transaction payloads apart from the blocks themselves, rendering the `0x46-0x4f` address space reserved for more transaction-related data that may be necessary in the future, such as the EOA's nonce.
+
+### Gas Costs
+
+The gas costs necessary for a particular instruction must weight the computational expense it brings to the system to assess a particular number. In the case of the instructions proposed by this EIP, this assessment was relatively straightforward given that identical instructions have already been introduced to the EVM and the instructions themselves would operate in an equivalent manner.
+
+### Instruction Space Pollution
+
+One can argue that multiple new EVM instructions pollute the EVM instruction address space and could cause issues in assigning sensible instruction codes to future instructions. This particular issue was assessed and a methodology via which the raw RLP encoded transaction may be accessible to the EVM was ideated. This would *future-proof* the new instruction set as it would be usable for other members of the transaction that may be desired to be accessible on-chain in the future, however, it would also cause a redundancy in the `ORIGIN` opcode.
+
+## Backwards Compatibility
+
+The EIP does not alter or adjust existing functionality provided by the EVM and as such, no known issues exist.
+
+## Test Cases
+
+TODO.
+
+## Reference Implementation
+
+A reference implementation will be provided for `go-ethereum` (Golang) which other implementations can be stemmed from should the EIP be moved to **`Accepted`** status.
+
```

Comment:
```suggestion
```
If you aren't planning on providing a reference implementation inline, just leave this section out.  Links to external repositories are strongly discouraged and usually not allowed, and in this case wouldn't be allowed.

Reply:
Should I also remove the Test Cases section in this case?

Reply:
If you don't think it is possible to have a set of reference test vectors, yes.  However, most opcode EIPs include a set of test vectors in the form of a table or JSON that are platform agnostic.  In this case, maybe a tiny little program (a dozen opcodes long) along with transaction input data.  Check out some of the other opcode EIPs to see examples of how people have done this in the past.

Reply:
I see, thank you. I will leave the Test Cases as `TODO` for now.